projects/addon-editor/components/editor-socket/editor-socket.component.ts

```diff
@@ -44,5 +47,32 @@ export class TuiEditorSocketComponent {
         @Optional()
         @Inject(TUI_SANITIZER)
         private readonly tuiSanitizer: Sanitizer | null,
+        @Inject(DOCUMENT)
+        private readonly document: Document,
+        @Optional()
+        @Inject(TuiTiptapEditorDirective)
+        private readonly editor: TuiTiptapEditorDirective | null,
     ) {}
+
+    /**
+     * @description:
+     * the main problem is that the external environment editor can use different base href="../"
+     * More information: https://rogerkeays.com/blog/using-base-href-with-anchors
+     */
+    @HostListener(`click`, [`$event`])
+    click(event: Event): void {
+        if (this.editor) {
+            return;
+        }
+
+        const href =
+            (event.target as HTMLElement)?.closest(`a`)?.getAttribute(`href`) ?? ``;
+
+        if (href.startsWith(`#`)) {
```

Comment:
nit: In our codestyle we do early returns if there's more than one line under `if` :)

Reply:
fixed